### PR TITLE
refactor(wasm-gen): Introduce generators for gear entry points and sys-calls

### PR DESCRIPTION
**Pull request by SabaunT**
The third planned PR with refactoring `wasm-gen`.

@gear-tech/dev

### CODE REVIEW

File: utils/wasm-gen/src/generator/syscalls/invocator.rs [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+            }
+        }
+
+        instructions.push(Instruction::Call(call_indexes_handle as u32));
+        instructions.extend(results);
+
+        Ok(instructions)
+    }
+
+    fn insert_sys_call_instructions(&mut self, instructions: &[Instruction]) -> Result<()> {
```
Comment (SabaunT) [inline utils/wasm-gen/src/generator/syscalls/invocator.rs:308]: @DennisInSky 

This doesn't skip insertion into internal function which calls properly `gr_reservation_send`. if you don't mind, will implement that in a separate PR.

File: utils/wasm-gen/src/generator/syscalls.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+pub use additional_data::*;
+pub use imports::*;
+pub use invocator::*;
+
+use gear_wasm_instrument::syscalls::{SysCallName, SysCallSignature};
+
+/// Type of invocable sys-call.
+///
+/// Basically, there are 2 types of generated sys-calls:
+/// 1. Those invocation of which is done regardless of validity of call context (``).
```
Comment (StackOverflowExcept1on) [inline utils/wasm-gen/src/generator/syscalls.rs:50]:
```suggestion
/// 1. Those invocation of which is done regardless of validity of call context (`Loose`).
```

File: utils/wasm-gen/src/generator/syscalls/invocator.rs (hunk 1 above)
Comment (SabaunT) [reply utils/wasm-gen/src/generator/syscalls/invocator.rs:308]: done in #3036

**Review by StackOverflowExcept1on — APPROVED** [review]

**Review by breathx — APPROVED** [review]

File: utils/wasm-gen/src/generator/syscalls.rs (hunk 2 above)
Comment (SabaunT) [reply utils/wasm-gen/src/generator/syscalls.rs]: done

**Pull request merged**